easybuild/tools/filetools.py

```diff
@@ -763,6 +763,13 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
+    # Attempt extracting the patch if it does not end in .patch (may end in .gz, .zip, etc.)
+    if not apatch.endswith(".patch"):
```

Comment:
this is a bit too generic, you should still require to have `.patch` in the filename, see my earlier remark about `patches` that can also contain files that are being copied rather than actual patches; see for example https://github.com/hpcugent/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/o/OpenBLAS/OpenBLAS-0.2.19-gompi-2016.09-LAPACK-3.6.1.eb#L35

This seems to be the cause of the other failing test (`test_extended_dry_run`)